test/System.ServiceModel.Federation.Tests/ClientCredentialsWrappingTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System.IdentityModel.Selectors;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using System.Threading;
+using Microsoft.IdentityModel.TestUtils;
+using Microsoft.IdentityModel.Tokens.Saml2;
+using Xunit;
+
+namespace System.ServiceModel.Federation.Tests
+{
+    public class ClientCredentialsWrappingTests
+    {
+        [Fact]
+        public void WrapsClientCredentials()
+        {
+            var issuedTokenParameters = new IssuedTokenParameters
+            {
+                IssuerAddress = new EndpointAddress(new Uri("https://localhost")),
+                IssuerBinding = new WSHttpBinding(SecurityMode.Transport),
+                SecurityKey = KeyingMaterial.RsaSecurityKey_1024,
+                Target = "https://localhost",
+                TokenType = Saml2Constants.OasisWssSaml2TokenProfile11
+            };
+
+            var binding = new CustomBinding(new WsFederationHttpBinding(issuedTokenParameters)
+            {
+                WSTrustContext = "DummyContext"
+            });
+            var clientCredentialCapturingBindingElement = new ClientCredentialCapturingBindingElement();
+            binding.Elements.Insert(1, clientCredentialCapturingBindingElement);
+
+            var clientCredentials = new ClientCredentials();
+            clientCredentials.UserName.UserName = "Foo";
+            clientCredentials.UserName.Password = "Bar";
+            var cf = binding.BuildChannelFactory<IRequestChannel>(clientCredentials);
```

Comment:
In general:
1. we have a much more comprehensive method for comparing if two objects are equal.
2. we use TheoryData that contains the expected value.
3. we stay away from [Fact] except in the simplest cases.
4. we stay away from Assert as only the first error is detected.

see: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/d5eac9cb8944f0ae8f1394efb06bbbd9b3fac052/test/Microsoft.IdentityModel.Protocols.WsTrust.Tests/RequestSecurityTokenResponseTests.cs#L62

We use the IdentityComparer here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/d5eac9cb8944f0ae8f1394efb06bbbd9b3fac052/test/Microsoft.IdentityModel.TestUtils/IdentityComparer.cs#L59

Your test should have a TheoryData derived from TheoryDataBase see: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/How-We-Test

You will need to add code similar to what was added for WsTrustRequest here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/d5eac9cb8944f0ae8f1394efb06bbbd9b3fac052/test/Microsoft.IdentityModel.TestUtils/IdentityComparer.cs#L180